### DISCUSSION TITLE
For rule apply

### DISCUSSION THREAD

**Discussion by rj10rd** [arendst/Tasmota#13622] [Q&A] [answered]
Rule1 On System#Boot do If (%time% > 22:00:00) power1 1 Endif If (%time% > 22:10:00) power1 0 Endif Endon

Is my rule right?

**Comment by barbudor** [comment, marked as answer]
No
Time is an integer
And I suggest using ELIF instead of 2 separated IF

**Reply by rj10rd** [reply]
By this rule my Frist command power1 1 work but second command power1 0 not working

**Reply by barbudor** [reply]
Just because you are trying now and %time% if > 22
Just look at the doc
%time% is number of minutes elapsed from midnight.

Beside your rule with 2 If is a bad idea (if it was working) because every minute after 22:10 it would power 1 and then power 0 very quickly

You should 1st compare to 22 * 60+10 to do power 0 and else compare to 22 * 60 to do power 1

You should also look at alternatives way of doing a 10 minutes pulse by using PulseTime
Or look also at Timers to configure a time for power on and another time for power off

**Reply by rj10rd** [reply]
On espeasy wiki i saw a example
On Pir#Switch=1 do
   If %systime% < 07:00:00
    Gpio,16,0
   Endif
   If %systime% > 19:00:00
    Gpio,16,1
   Endif
  Endon
In tasmota what i use beside %systime%

**Reply by sfromis** [reply]
Since it is a boot rule, it will not trigger every minute, but only act once at boot time. I suspect that this scenario may not be what you want, but without any details, it is hard to tell in how many ways it is not correct :-)

**Reply by barbudor** [reply]
Indeed 🤣

**Reply by rj10rd** [reply]
So please tell me what Rule Variables for system Time

**Reply by Jason2866** [reply]
It is written here https://tasmota.github.io/docs/Rules/

**Reply by barbudor** [reply]
You already know it
You just not comparing correctly

**Reply by rj10rd** [reply]
> It is written here https://tasmota.github.io/docs/Rules/

LOCALTIME rule variable is correct for me?

**Reply by barbudor** [reply]
![image](https://user-images.githubusercontent.com/2996042/140988908-e5049ab9-00bd-47dc-b7e6-f855c6ae04c6.png)

22:00:00 = 22 * 60 = 1320
22:10:00 = 22 * 60 + 10 = 1330

**Reply by rj10rd** [reply]
> ![image](https://user-images.githubusercontent.com/2996042/140988908-e5049ab9-00bd-47dc-b7e6-f855c6ae04c6.png)
> 
> 22:00:00 = 22 * 60 = 1320 22:10:00 = 22 * 60 + 10 = 1330

Thank for nice explanation
But my problem is that
I want trun on water heater on morning time between 4 to 6 am
And on my place power come in morning time between 4 to 6 am
So i want when tasmota boot between 4 to 6 am it start water heater.
Sorry for my weak English.

**Reply by sfromis** [reply]
As long as you use the right type of time comparison (with minutes instead of hh:mm:ss, the structure of your original suggestion looks ok, assuming that you're using a binary including IF/ENDIF

04:00 would be minutes 240, and 06:00 would be minutes 360.

**Reply by rj10rd** [reply]
I want start water heater after 4am and trun off after 6am so for this

Rule1 On System#Boot do If (%time% > 240) power1 1 Endif If (%time% > 360) power1 0 Endif Endon
Correct?

**Reply by ascillato** [reply]
Hi, sorry, that rule won't work. ON SYSTEM#BOOT is triggered ONLY just ONCE after you powered up your device. If your device has energy all the time that won't trigger anything. Besides that, at the moment that Tasmota triggers ON SYSTEM#BOOT, your device is not connected yet to Wi-Fi so Tasmota don't have time. So that rule has a trigger that is FALSE on every Tasmota START UP.

Seems that you want just to POWER ON the relay 1 at 4 am and POWER OFF the relay 1 at 6 am, but EVERYDAY right?

In that case, you don't need any rule, nor any custom version of Tasmota. You just need to go to TIMERS in the Tasmota Web UI and set a timer to turn on RELAY 1 at 4 AM and another timer to turn OFF the relay at 6 AM.

![image](https://user-images.githubusercontent.com/35405447/84083542-38307d80-a9b8-11ea-8e6b-fa86a92937ce.png)

**Reply by rj10rd** [reply]
Thanks for reply sir but problem is that not exactly 4am. Between 4am to 6am and after 6am trunoff Power1.

**Reply by ascillato** [reply]
What do you mean? Your device has energy all the time or not?

**Reply by rj10rd** [reply]
No sir. My device doesn't have energy all time. That's the real problem. Energy can came between 4 to 6am.

**Reply by rj10rd** [reply]
> Just because you are trying now and %time% if > 22 Just look at the doc %time% is number of minutes elapsed from midnight.
> 
> Beside your rule with 2 If is a bad idea (if it was working) because every minute after 22:10 it would power 1 and then power 0 very quickly
> 
> You should 1st compare to 22 * 60+10 to do power 0 and else compare to 22 * 60 to do power 1
> 
> You should also look at alternatives way of doing a 10 minutes pulse by using PulseTime Or look also at Timers to configure a time for power on and another time for power off

Rule2 On System#Boot do If (%time%>748) power1 0 ELSE (%time%>720) power1 1 Endif Endon
am i right?

**Reply by rj10rd** [reply]
@barbudor  thank you sir
you are excellent

**Reply by ascillato** [reply]
The trigger `ON system#boot DO` don't have the time at that moment. For your use-case it is better to use `ON Time#Initialized DO` because with that one, you will ensure that Tasmota will check for the time AFTER Tasmota has synced the time.

**Reply by barbudor** [reply]
Regarding the syntax you should use ELIF shortcut for ELSE IF:
Rule2 On System#Boot do If (%time%>748) power1 0 ELIF (%time%>720) power1 1 Endif Endon
